### PR TITLE
5.0.0.16

### PR DESCRIPTION
**Pull request by Ghost-chu**

### CODE REVIEW

**Comment by Ghost-chu** [comment]
Fixed caching system, Sign hex color bug and hopper updating

File: src/main/java/org/maxgamer/quickshop/util/Util.java [hunk 1]
```diff
@@ -1110,6 +1110,7 @@ public static Material getSignMaterial() {
      * @return Json
      */
     public static String componentsToJson(BaseComponent[] components) {
+        plugin.getLogger().info(ComponentSerializer.toString(components));
```
Comment (sandtechnology) [inline src/main/java/org/maxgamer/quickshop/util/Util.java:1113]: Please remove debug code

File: src/main/java/org/maxgamer/quickshop/util/Util.java [hunk 2]
```diff
@@ -1330,4 +1331,19 @@ public static String createRegexFromGlob(@NotNull String glob) {
...
+     * @param loc location
+     * @return blocked location
+     */
+    @NotNull
+    public static Location getBlockLocation(@NotNull Location loc) {
+        loc = loc.clone();
+        loc.setX(loc.getBlockX());
+        loc.setY(loc.getBlockY());
+        loc.setZ(loc.getBlockZ());
+        return loc;
```
Comment (sandtechnology) [inline src/main/java/org/maxgamer/quickshop/util/Util.java:1346]: What about  just ```return new Location(loc.getWorld(),loc.getBlockX(),loc.getBlockY(),loc.getBlockZ());``` ?

**Review by sandtechnology — CHANGES_REQUESTED** [review]

File: src/main/java/org/maxgamer/quickshop/util/Util.java (hunk 2 above)
Comment (Ghost-chu) [reply src/main/java/org/maxgamer/quickshop/util/Util.java:1345]: may lose pitch and yaw

Comment (sandtechnology) [reply src/main/java/org/maxgamer/quickshop/util/Util.java:1345]: Just a idea, if you want to use it, just add pitch and yaw in there aswell

**Pull request merged**